plugins/eventMacro/eventMacro/Core.pm

```diff
@@ -195,22 +195,29 @@ sub create_automacro_list {
 		}
 		
 		PARAMETER: foreach my $parameter (@{$value->{'parameters'}}) {
-		
 			###Check Duplicate Parameter
 			if (exists $currentParameters{$parameter->{'key'}}) {
 				warning "[eventMacro] Ignoring automacro '$name' (parameter ".$parameter->{'key'}." duplicate)\n";
 				next AUTOMACRO;
 			}
-			###Parameter: call
-			if ($parameter->{'key'} eq "call" && !$self->{Macro_List}->getByName($parameter->{'value'})) {
-				warning "[eventMacro] Ignoring automacro '$name' (call '".$parameter->{'value'}."' is not a valid macro name)\n";
-				next AUTOMACRO;
-			
+			###Parameter: call with or without param
+			if ($parameter->{'key'} eq "call" && $parameter->{'value'} =~ /(\S+)\s+(.*)?/) {
+				if (!$self->{Macro_List}->getByName($1) ) {
```

Comment:
It's not safe to use regex capture variables (eg, `$1`, `$2`) after calling another function. If that function does a regex anywhere, it'll overwrite the capture variables. So this call is safe, but additional uses of `$1` in this code are not.

While this might not be a problem right now, if somebody changes `getByName` to include a regex, it'll break this code totally unexpectedly. Please store the capture variables in temporary variables if you need to use them more after calling a function.
```
my ($name, $variables) = ($1, $2);

Reply:
i didn't know that, i'll update the code, thanks for the tip